src/components/application_manager/test/policy_handler_test.cc

```diff
@@ -1467,24 +1641,929 @@ TEST_F(PolicyHandlerTest, SendMessageToSDK) {
+  EnablePolicyAndPolicyManagerMock();
+  utils::SharedPtr<application_manager_test::MockApplication> invalid_app;
+  EXPECT_CALL(app_manager_, active_application()).WillOnce(Return(invalid_app));
+
+  EXPECT_FALSE(policy_handler_.SendMessageToSDK(msg, url));
+}
+#endif
 TEST_F(PolicyHandlerTest, CanUpdate) {
   GetAppIDForSending();
   EXPECT_TRUE(policy_handler_.CanUpdate());
 }
 
+TEST_F(PolicyHandlerTest, CanUpdate_TwoApplicationForSending_SUCCESS) {
+  EnablePolicyAndPolicyManagerMock();
+  utils::SharedPtr<application_manager_test::MockApplication> second_mock_app =
+      utils::MakeShared<application_manager_test::MockApplication>();
+
+  EXPECT_CALL(*mock_app_, hmi_level())
+      .WillOnce(Return(mobile_apis::HMILevel::HMI_FULL))
+      .WillOnce(Return(mobile_apis::HMILevel::HMI_LIMITED));
+  EXPECT_CALL(*second_mock_app, hmi_level())
+      .WillRepeatedly(Return(mobile_apis::HMILevel::HMI_LIMITED));
+
+  EXPECT_CALL(*second_mock_app, app_id()).WillRepeatedly(Return(kAppId2_));
+  EXPECT_CALL(*mock_app_, app_id()).WillRepeatedly(Return(kAppId1_));
+
+  // Check expectations
+  EXPECT_CALL(*mock_app_, IsRegistered()).WillOnce(Return(true));
+  test_app.insert(mock_app_);
+  test_app.insert(second_mock_app);
+
+  EXPECT_CALL(mock_session_observer, GetDataOnDeviceID(_, _, _, _, _))
+      .WillOnce(DoAll(SetArgPointee<3>(kMacAddr_), Return(0)));
+
+  EXPECT_CALL(*mock_policy_manager_, GetUserConsentForDevice(kMacAddr_))
+      .WillRepeatedly(Return(kDeviceAllowed));
+
+  EXPECT_TRUE(policy_handler_.CanUpdate());
+}
+
+namespace {
+/**
+ * @brief The WaitAsync class
+ * can wait for a certain amount of function calls from different
+ * threads, or a timeout expires.
+ */
+class WaitAsync {
+ public:
+  WaitAsync(const uint32_t count, const uint32_t timeout)
+      : count_(count), timeout_(timeout) {}
+
+  void Notify() {
+    count_--;
+    cond_var_.NotifyOne();
+  }
+
+  bool Wait(sync_primitives::AutoLock& auto_lock) {
+    while (count_ > 0) {
+      sync_primitives::ConditionalVariable::WaitStatus wait_status =
+          cond_var_.WaitFor(auto_lock, timeout_);
+      if (wait_status == sync_primitives::ConditionalVariable::kTimeout) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+ private:
+  int count_;
+  const uint32_t timeout_;
+  sync_primitives::ConditionalVariable cond_var_;
+};
+}
+
+ACTION_P(NotifyAsync, waiter) {
+  waiter->Notify();
+}
+
+TEST_F(PolicyHandlerTest,
+       OnAppPermissionConsentInternal_ValidConnectionKey_SUCCESS) {
+  // EnablePolicyAndPolicyManagerMock();
+
+  ChangePolicyManagerToMock();
+  const uint32_t device = 2u;
+
+  PermissionConsent permissions;
+  permissions.device_id = kDeviceId_;
+  permissions.consent_source = "consent_source";
+
+  policy::FunctionalGroupPermission group_permission_allowed;
+  group_permission_allowed.state = GroupConsent::kGroupAllowed;
+  group_permission_allowed.group_alias = "allowed";
+  group_permission_allowed.group_name = "name_allowed";
```

Comment:
@okozlovlux Use here method which you will create for filling FunctionalGroupPermission